app/src/views/home/home_screen.tsx

```diff
@@ -119,7 +131,11 @@ export default function HomeScreen({
           >
             <ThemedTextButton
               style={styles.createGroupButton}
-              onPress={() => navigation.navigate("CreateGroup")}
+              onPress={() =>
+                userIsGuest
+                  ? handleRestrictedAccess("CreateGroup")
+                  : navigation.navigate("CreateGroup")
+              }
```

Comment:
Why didn't you just use handleRestrictedAccess ? It already checks if user is guest in it

Reply:
Yes you're right good catch !